src/apprt/gtk/menu.zig

```diff
@@ -0,0 +1,136 @@
+
+            const menu_widget: *MenuWidget = switch (style) {
+                .popover_menu, .popover_menu_no_arrow => brk: {
+                    const menu_widget: *MenuWidget = MenuWidget.newFromModelFull(menu_model, .{ .nested = true });
+                    if (style == .popover_menu_no_arrow) {
+                        menu_widget.as(gtk.Popover).setHasArrow(0);
+                    }
+                    _ = gtk.Popover.signals.closed.connect(
+                        menu_widget,
+                        *Self,
+                        gtkRefocusTerm,
+                        self,
+                        .{},
+                    );
+                    break :brk menu_widget;
+                },
+                .popover_menu_bar => brk: {
+                    break :brk gtk.PopoverMenuBar.newFromModel(menu_model);
+                },
+            };
+
+            self.* = .{
+                .parent = parent,
+                .menu_widget = menu_widget,
+            };
+        }
+
+        pub fn setParent(self: *const Self, widget: *c.GtkWidget) void {
+            self.menu_widget.as(gtk.Widget).setParent(@ptrCast(@alignCast(widget)));
+        }
+
+        pub fn asWidget(self: *const Self) *c.GtkWidget {
+            return @ptrCast(@alignCast(self.menu_widget));
+        }
+
+        pub fn isVisible(self: *const Self) bool {
+            return self.menu_widget.as(gtk.Widget).getVisible() != 0;
+        }
+
+        pub fn setVisible(self: *const Self, visible: bool) void {
+            self.menu_widget.as(gtk.Widget).setVisible(@intFromBool(visible));
+        }
+
+        pub fn refresh(self: *const Self) void {
+            const window: *gtk.Window, const has_selection: bool = switch (T) {
+                Window => window: {
+                    const core_surface = self.parent.actionSurface() orelse break :window .{
+                        @ptrCast(@alignCast(self.parent.window)),
+                        false,
+                    };
+                    const has_selection = core_surface.hasSelection();
+                    break :window .{ @ptrCast(@alignCast(self.parent.window)), has_selection };
+                },
+                Surface => surface: {
+                    const window = self.parent.container.window() orelse return;
+                    const has_selection = self.parent.core_surface.hasSelection();
+                    break :surface .{ @ptrCast(@alignCast(window.window)), has_selection };
+                },
+                else => unreachable,
+            };
+
+            const action_map: *gio.ActionMap = @ptrCast(@alignCast(window));
+            const action: *gio.SimpleAction = @ptrCast(@alignCast(action_map.lookupAction("copy") orelse return));
+            action.setEnabled(@intFromBool(has_selection));
+        }
+
+        pub fn popupAt(self: *const Self, x: f64, y: f64) void {
+            const rect: gdk.Rectangle = .{
+                .f_x = @intFromFloat(x),
```

Comment:
We should let the caller convert the float to an int on their side. Accepting a floating point value and immediately converting it to an integer is a bit disingenuous.

Reply:
Done.